Very big change (read below)

Leone25 opened the pull request:
- major code restructurezation, all the music handaling functions got moved to `player.js`, now all the available sources are in `src/commands/sources/` for ease of implementation and cleaner code
- youtube play back reverted to `ytdl-core`, modification temporary untill i figure out how to not make youtube cut my connection and stuff
- implemented propper search command, doing `!search [query]` will return a 10 point list from which the user can select with `!play [n]` to play the corrispective song

everything mostly tested, tho might still be flew errors and stuff, wouldn't be surprised it there is some big ones i missed

Leone25 committed:
Very big change (read below)

major code restructurezation, all the music handaling functions got moved to `player.js`, now all the available sources are in `src/commands/sources/` for ease of implementation and cleaner code
youtube play back reverted to `ytdl-core`, modification temporary untill i figure out how to not make youtube cut my connection and stuff
implemented propper search command, doing `!search [query]` will return a 10 point list from which the user can select with `!play [n]` to play the corrispective song

everything mostly tested, tho might still be flew errors and stuff, wouldn't be surprised it there is some big ones i missed

Leone25 committed:
Queue bug fix

Leone25 committed:
queue bug fix

Leone25 commented:
The last one is actually is a loop bug fix

zacwm committed:
Fix search range

Fix for selecting the 10th search result instead of letting it pass to the "Invalid range" message.

zacwm commented on src/commands/sources/soundcloud.js:
```diff
@@ -0,0 +1,61 @@
+
+exports.getInfo = (url) => {
+    return new Promise(async (resolve, reject) => {
+        youtubedl.getInfo(url, [], function(err, info) {
```
https://user-images.githubusercontent.com/17349626/103175146-830dd600-48b3-11eb-9937-0f859d6a3d2f.mp4

There seems to be an issue with `youtubedl.getInfo()`

After some testing on this issue, **it does call the SoundCloud file!** So it appears to be an issue with this module. **I've uploaded a video displaying this...**

After requesting a SoundCloud track URL, it will instead search on YouTube for some similar titled video as the SoundCloud track.

I'm unsure with youtubedl on how to fix this, but maybe a SoundCloud API request is possible too...

Leone25 committed:
Removed `youtube-dl`

removed `youtube-dl` by implementing alternative solutions mostly resolving to original api

also fixed bad implementation of specification for sources

Leone25 committed:
solved bug
